API/Controllers/SearchController.cs

```diff
@@ -0,0 +1,58 @@
+using API.DTOs;
+using API.Services.Interfaces;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[EnableCors]
+[Authorize(Roles = "system")] // Internal access only at this time
+[AllowAnonymous]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class SearchController(ITournamentsService tournamentsService, IMatchesService matchesService, IPlayerService playerService) : Controller
+{
+    [HttpPost]
+    public async Task<IActionResult> SearchByNames([FromQuery] string? tournamentName, [FromQuery] string? matchName, [FromQuery] string? username)
```

Comment:
Method return type should be concrete (e.g. `ActionResult<T>`)

Reply:
Good catch, came back to this just specifically to check but you already got it hehe

Reply:
I think we decided to return ActionResults and then specify the return type in a ProducesResponseType attribute, I added the attribute in a commit I'll be pushing soon.

Reply:
@myssto Honestly can't remember if we defined this concretely. What do you think about returning `Task<ActionResult<T>>` *and* documenting the `[ProducesResponseType]` attribute? Can you look and see if swagger can detect just from the ActionResult<T>?

Reply:
Will look into it 👍

Reply:
![image](https://github.com/osu-tournament-rating/otr-api/assets/41877052/97af23c9-b65a-4f6a-a30a-40fadeed600e)

Looks like swagger will pickup the return object from the `[ProducesResponseType]` attribute

Reply:
I actually found a decent solution to this last night. Take a look at #136 and #207 for a more in depth explanation. The short of it is that we should be using a return type of the new `TypedResponse<T>`

Reply:
Alright will refactor this endpoint to comply with the new standard we're using

Reply:
Updated return types